src/ansys/simai/core/data/model_configuration.py

```diff
@@ -0,0 +1,257 @@
+
+
+@dataclass
+class ModelConfiguration:
+    """Configures the build of a model.
+
+    Args:
+        project: the project of the configuration.
+        build_preset: indicates the build duration. Available options:
+
+                    | *debug*: < 30 min
+
+                    | *short*: < 24 hours
+
+                    | *default*: < 2 days, default value.
+
+                    | *long*: < 1 week
+        continuous: indicates if continuous learning is enabled. Default is False.
+        input: the inputs of the model.
+        output: the outputs of the model.
+        global_coefficients: the Global Coefficients of the model.
+
+    Example:
+        Define a new configuration and launch a build.
+
+        .. code-block:: python
+
+            import ansys.simai.core as asc
+            from ansys.simai.core.data.model_configuration import (
+                ModelConfiguration,
+                ModelInput,
+                ModelOutput,
+            )
+
+            simai = asc.from_config()
+
+            # Get the project of interest
+            aero_dyn_project = simai.projects.get(name="aero-dyn")
+
+            # Define the inputs of the model
+            model_input = ModelInput(surface=["Velocity"], boundary_conditions=["Vx"])
+
+            # Define the outputs of the model
+            model_output = ModelOutput(
+                surface=["Pressure", "WallShearStress_0"], volume=["Velocity_0", "Pressure"]
+            )
+
+            # Define the model coefficient
+            global_coefficients = [("max(Pressure)", "maxpress")]
+
+            # Define the build configuration for the model
+            new_conf = ModelConfiguration(
+                project=aero_dyn_project,
+                build_preset="debug",
+                continuous=False,
+                input=model_input,
+                output=model_output,
+                global_coefficients=global_coefficients,
+                simulation_volume=simulation_volume,
+            )
+
+            # Launch a mode build with the new configuration
+            new_model = simai.models.build(new_conf)
+    """
+
+    project: "Project" = None
+    build_preset: Literal[
+        "debug",
+        "short",
+        "default",
+        "long",
+    ] = "default"
+    continuous: bool = False
+    input: ModelInput = ModelInput()
+    output: ModelOutput = ModelOutput()
+
+    def __set_gc(self, gcs: list[tuple | GlobalCoefficientUnit]):
+        verified_gcs = []
+        for gc in gcs:
+            gc_unit = GlobalCoefficientUnit(*gc) if isinstance(gc, tuple) else gc
+            self.project.verify_gc_formula(
+                gc_unit.formula, self.input.boundary_conditions, self.output.surface
+            )
+            verified_gcs += [gc_unit]
+        self.__dict__["global_coefficients"] = verified_gcs
+
+    def __get_gc(self):
+        return self.__dict__.get("global_coefficients")
+
+    global_coefficients: ModelOutput = property(__get_gc, __set_gc)
```

Comment:
```suggestion
    global_coefficients: Optional[list[GlobalCoefficientUnit]] = property(__get_gc, __set_gc)
```
^ remove optional if you default to an empty list instead of None
Btw it's not pretty but seems like we're limited by what dataclass can do here